### PR TITLE
reloadable gem + proof of concept

### PR DESCRIPTION
**Pull request by phil-monroe**

### CODE REVIEW

File: dom_control/lib/dom_control.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module DomControl
```
Comment (phil-monroe) [inline dom_control/lib/dom_control.rb:1]: would love to actually change the namespace to`DOMControl`, but ran into autoloading issues with previous attempts getting this working, but I think the `Zeitwerk::Registry` approach should now make it naming agnostic... probably...

File: config/environments/development.rb [hunk 2]
```diff
@@ -1,6 +1,40 @@
...
+    Rails.application.reloaders << Class.new do
+      def initialize(file_watcher)
+        @file_watcher = file_watcher
+      end
+
+      def updated?
+        @file_watcher.execute_if_updated
+      end
+    end.new(file_watcher)
+  end
```
Comment (phil-monroe) [inline config/environments/development.rb:33]: mildly more annoying than I remember this being, but generally makes sense now that it is working.

**Comment by phil-monroe** [comment]
@rickychilcott - had fun playing around with the gem tonight and honestly feel like it could be most of the way there... (minus specs and whatnot)

**Review by rickychilcott — APPROVED** [review]
👏👏👏

Thank you, Phil!

**Pull request merged**